File: leetcode-solutions/ruby/283. Move Zeroes.rb
# https://leetcode.com/problems/move-zeroes/

# @param {Integer[]} nums
# @return {Void} Do not return anything, modify nums in-place instead.
# two pointer solution, 
# start with both zero but one will increment everytime but other will
# only non zero found
def move_zeroes(nums)
  left = 0
  right = 0
  
  while right < nums.size
      if nums[right] != 0
          nums[left], nums[right] = nums[right], nums[left]
          left += 1
      end
      
      right += 1
  end
end

File: leetcode-solutions/python/Algorithm-1/283. Move Zeroes.py
# https://leetcode.com/problems/move-zeroes/
# using 2 pointers
class Solution:
    def moveZeroes(self, nums: List[int]) -> None:
        """
        Do not return anything, modify nums in-place instead.
        """
        start, current, size = 0, 0, len(nums)
        while current < size:
            if nums[current] != 0:
                nums[start], nums[current] = nums[current], nums[start]
                start += 1
                current += 1
            else:
                current += 1